### PR TITLE
Cleaning up index.js

### PR DESCRIPTION
**Pull request by Coenego**

### CODE REVIEW

File: shared/gh/js/views/gh.header.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+define(['gh.core', 'gh.login-form', 'gh.subheader'], function(gh) {
+
+    /**
+     * Set up the header component by rendering the header and login templates, fetching the tripos
```
Comment (bp323) [inline shared/gh/js/views/gh.header.js]: `fetching the tripos` structure has already happened at this point

File: shared/gh/js/views/gh.header.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        gh.utils.renderTemplate('header', {
+            'data': {
+                'gh': gh
+            }
+        }, $('#gh-header'), function() {
+            // Render the tripos pickers
+            gh.utils.renderTemplate('subheader-pickers', {
+                'gh': gh
+            }, $('#gh-subheader'), function() {
+                // Set up the tripos picker after all data has been retrieved
```
Comment (bp323) [inline shared/gh/js/views/gh.header.js]: Should be reworded as data has already been fetched

File: shared/gh/js/views/gh.login-form.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            var formValues = _.object(_.map($(this).serializeArray(), _.values));
+            gh.api.authenticationAPI.login(formValues.username, formValues.password, function(err) {
+                if (err) {
+                    return gh.utils.notification('Could not sign you in', 'Please check that you are entering a correct username & password', 'error');
+                }
+                window.location.reload();
+            });
+
+            // Avoid default form submit behaviour
+            return false;
```
Comment (bp323) [inline shared/gh/js/views/gh.login-form.js]: Should be return outside of the if?

**Comment by bp323** [comment]
I'll have another pass on this, not sure if it's ready for merge?

**Comment by Coenego** [comment]
Assigning to @bp323 for review

**Comment by bp323** [comment]
Back to @Coenego for followup on the calendar issue

**Comment by Coenego** [comment]
Assigning to @bp323 for review after fixing issue

**Comment by Coenego** [comment]
Back to @bp323 for review

**Comment by bp323** [comment]
Verified and merged

**Pull request merged**